src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImageFunctionalTest.groovy

```diff
@@ -0,0 +1,99 @@
+package com.bmuschko.gradle.docker.tasks.image
+
+import com.bmuschko.gradle.docker.AbstractFunctionalTest
+import org.gradle.testkit.runner.BuildResult
+
+class DockerSaveImageFunctionalTest extends AbstractFunctionalTest {
+
+    final String ImageId = createUniqueImageId()
+    final String controlSavedImage = "build/docker/${ImageId}-docker-image-control.tar"
+
+    def setup() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task dockerfile(type: Dockerfile) {
+                from '$TEST_IMAGE_WITH_TAG'
+            }
+
+            task buildImage(type: DockerBuildImage) {
+                dependsOn dockerfile
+                inputDir = file("build/docker")
+                tag = "${ImageId}"
+                labels = ["setup":"${ImageId}"]
+            }
+            task saveImageControl(type: DockerSaveImage) {
+                dependsOn buildImage
+                tag = "${ImageId}"
+                repository = "${ImageId}"
+                destFile = file("${controlSavedImage}")
+            }            
+        """
+
+        when:
+        BuildResult result = build('saveImageControl')
+
+        then:
+        result.output.contains("Created image with ID")
+    }
+
+    def "Can docker image be saved without compression"() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn buildImage
+                tag = "${ImageId}"
+                repository = "${ImageId}"
+                destFile = file("build/docker/${ImageId}-docker-image.tar")
+            }
+        """
+
+        when:
+        build('saveImage')
+
+        then:
+        noExceptionThrown()
+        fileSizesMatch(new File(projectDir, "build/docker/${ImageId}-docker-image.tar"), new File(projectDir, controlSavedImage))
```

Comment:
why do you compare size of the same file? Or am I missing something?

Reply:
It's not the same file, it's the result of the same task being called twice.

What is the expectation of unpacking the tar of a docker image?

Reply:
OK, now I see how it works.

Reply:
It's simple and crude.  Regarding your comment above:
>In functional test I would rather test if archive can be unpacked and read.

Do you have some insight about how I could verify the saved image based on something from docker?  Or is the expectation that it is already handled by the upstream docker-java project?